<launch>

	<arg name="legs_odom_topic_name" default="legs_odom"/>
	<arg name="base_frame" default="base_footprint"/>
    <arg name="imu" default="true"/>

	
	<!-- Legs odometry -->

	<rosparam command="load" file="$(find clhero_odom)/config/clhero_geom.yaml" />
	<node name="clhero_odom" pkg="clhero_odom" type="clhero_odom_no_tf" output="screen">
		<remap from="odom" to="$(arg legs_odom_topic_name)"/>
        <param name="imu" value="$(arg imu)"/>
	</node>

    <!--IMU-->

    <include file="$(find razor_imu_9dof)/launch/razor-pub.launch"/>

    <!--realsense-->

    <include file="$(find realsense2_camera)/launch/rs_camera_depth_rectifier.launch"/>

    <!--node name="depthimage_to_laserscan" pkg="depthimage_to_laserscan" type="depthimage_to_laserscan">
        <remap from="image" to="/camera/aligned_depth_to_color/image_raw"/>
        <param name="range_min" value="0.05"/>
    </node-->

    <include file="$(find hokuyo_node)/hokuyo_test.launch"/>

    <!--Orbslam-->

    <!--node pkg="ORB_SLAM2" type="RGBD" name="orbslam_odom" args="ws/clhero_ws/src/ORB_SLAM2/Vocabulary/ORBvoc.txt ws/clhero_ws/src/realsense-ros/realsense2_camera/config/realsense_640.yaml"/-->


	<!-- EKF -->

	<node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
    	<param name="output_frame" value="odom"/>
    	<param name="freq" value="30.0"/>
    	<param name="sensor_timeout" value="2.0"/>
    	<param name="odom_used" value="true"/>
    	<param name="imu_used" value="$(arg imu)"/>
    	<param name="vo_used" value="false"/>
    	<param name="debug" value="false"/>
    	<param name="self_diagnose" value="false"/>
    	<remap from="odom" to="$(arg legs_odom_topic_name)"/>
    	<remap from="robot_pose_ekf/odom_combined" to="/odom_combined"/>
        <remap from="vo" to="orb_slam/odom"/>
  	</node>

    <node pkg="clhero_localization" type="odom_filler" name="odom_filler">
        <param name="base_frame" value="$(arg base_frame)"/>
    </node>


</launch>
